### PR TITLE
[Task#68292] Add error handling logic and render result to UI

### PR DESCRIPTION
**Pull request by tungbd-0214**
## Related Tickets:
- [#68292](https://edu-redmine.sun-asterisk.vn/issues/68292)

## WHAT this PR does?
Add error handling logic and render result to UI

## Notes Impacted Areas:
- N/A.

## Attachments:

Screen Shot 2023-09-17 at 17 00 42

### CODE REVIEW

**Review by Hungnd-0370 — APPROVED** [review]
oke

**Comment by tungbd-0214** [comment]
ready

File: app/Http/Controllers/RegistrationController.php [hunk 1]
```diff
@@ -17,4 +17,19 @@ public function index()
             ]
         );
     }
+
+    public function reject(Request $request, Registration $registration)
+    {
+        try {
+            $registration->update(['status'=> 2]);
```
Comment (ghost) [inline app/Http/Controllers/RegistrationController.php]: what is 2? chuyển 2 vào config

File: app/Http/Controllers/RegistrationController.php [hunk 2]
```diff
@@ -17,4 +17,19 @@ public function index()
...
+        } catch (Exception $e) {
+            return redirect()->back()->withErrors(
+                [
+                    'reject' => 'there was an error'
+                ]
+            );
```
Comment (ghost) [inline app/Http/Controllers/RegistrationController.php]: localization
```suggestion
                [
                    'reject' => __('there was an error'),
                ]
            );
```

File: app/Http/Controllers/SalonController.php [hunk 3]
```diff
@@ -48,26 +48,34 @@ public function store(StoreSalonRequest $request)
...
+                function () use ($validated) {
+                    DB::table('salons')->insert(
+                        [
+                        'owner_email' => $validated['email'],
+                        'name' => $validated['salon_name'],
+                        'address' => $validated['address'],
+                        'package_id' => $validated['package_id'],
+                        'is_active' => true,
+                        'created_at' => now(),
+                        'updated_at' => now()
```
Comment (ghost) [inline app/Http/Controllers/SalonController.php]:
```suggestion
                        'updated_at' => now(),
```

File: app/Http/Controllers/SalonController.php [hunk 4]
```diff
@@ -48,26 +48,34 @@ public function store(StoreSalonRequest $request)
...
+                        ]
+                    );
+
+                    DB::table('registrations')->where('email', $validated['email'])->update([ 'status' => 1]);
```
Comment (ghost) [inline app/Http/Controllers/SalonController.php]:
```suggestion
                    DB::table('registrations')->where('email', $validated['email'])->update(['status' => config(...)]);
```

File: app/Http/Controllers/SalonController.php [hunk 5]
```diff
@@ -48,26 +48,34 @@ public function store(StoreSalonRequest $request)
...
+                },
+                config('database.connections.mysql.max_attempts')
+            );
+        } catch (Exception $e) {
+            return redirect()->back()->withErrors(
+                [
+                    'store' => 'ups, there was an error'
+                ]
+            );
+        }
```
Comment (ghost) [inline app/Http/Controllers/SalonController.php]: localization

File: resources/js/Layouts/Authenticated.jsx [hunk 6]
```diff
@@ -183,6 +183,7 @@ export default function Authenticated({ auth, children }) {
                             padding: 24,
                             minHeight: 280,
                             background: '#fff',
+                            overflow: 'scroll'
```
Comment (ghost) [inline resources/js/Layouts/Authenticated.jsx]:
```suggestion
                            overflow: 'scroll',
```

File: resources/js/Pages/Registrations.jsx [hunk 7]
```diff
@@ -1,25 +1,80 @@
...
+
+    const accept = (e, record) => {
+
+        e.preventDefault();
+
+        Inertia.post('/salons', record, {
+            onSuccess: () => {
+                openNotification('success',
+                    'Successfully accepted',
+                    'Accepted salon is added to the system.'
```
Comment (ghost) [inline resources/js/Pages/Registrations.jsx]: localization

File: resources/js/Pages/Registrations.jsx [hunk 8]
```diff
@@ -48,26 +103,61 @@ export default function Registrations(props) {
...
+                {
+                    text: 'Rejected',
+                    value: 2,
+                },
+            ],
+            onFilter: (value, record) => record.status === value,
             render: (text, record) => {
                 return (
-                    <form onSubmit={(e) => { submit(e, record) }}>
+                    record.status === 1 ?
```
Comment (ghost) [inline resources/js/Pages/Registrations.jsx]: 1???

**Comment by tungbd-0214** [comment]
ready

**Comment by tungbd-0214** [comment]
ready

File: app/Http/Controllers/RegistrationController.php [hunk 9]
```diff
@@ -10,11 +10,35 @@ class RegistrationController extends Controller
...
+
+        foreach ($registrations as $registration) {
+            $registration->status = config('app.registration_status')[$registration->status];
+        }
+
         return Inertia::render(
             'Registrations',
             [
-            'registrations' => Registration::all(),
+            'registrations' => $registrations,
```
Comment (ghost) [inline app/Http/Controllers/RegistrationController.php]:
```suggestion
                'registrations' => $registrations,
```

File: app/Http/Controllers/SalonController.php [hunk 10]
```diff
@@ -48,26 +48,38 @@ public function store(StoreSalonRequest $request)
...
+                    DB::table('salons')->insert(
+                        [
+                        'owner_email' => $validated['email'],
+                        'name' => $validated['salon_name'],
+                        'address' => $validated['address'],
+                        'package_id' => $validated['package_id'],
+                        'is_active' => true,
+                        'created_at' => now(),
+                        'updated_at' => now(),
+                        ]
```
Comment (ghost) [inline app/Http/Controllers/SalonController.php]:
```suggestion
                        [
                            'owner_email' => $validated['email'],
                            'name' => $validated['salon_name'],
                            'address' => $validated['address'],
                            'package_id' => $validated['package_id'],
                            'is_active' => true,
                            'created_at' => now(),
                            'updated_at' => now(),
                        ]
```

File: resources/js/Pages/Registrations.jsx [hunk 11]
```diff
@@ -77,21 +167,46 @@ export default function Registrations(props) {
...
             <div className="py-12">
+                <div className='sm:px-6 lg:px-8 w-full'>
+                    <h2 className='font-semibold text-2xl text-gray-800 leading-tight'>{lang.get('strings.Registrations')}</h2>
+                </div>
+                <div className="flex justify-end w-full mr:3 mb-8 sm:px-6 lg:px-8">
+                    <Search placeholder="input id, salon name, address"
+                        enterButton bordered
+                        size="large"
+                        allowClear
+                        onChange={searchChangeHandler}
```
Comment (ghost) [inline resources/js/Pages/Registrations.jsx]:
```suggestion
                        onChange={ searchChangeHandler }
```

File: routes/web.php [hunk 12]
```diff
@@ -44,6 +44,14 @@ function () {
             'index'
             ]
         )->name('registrations.index');
+
+        Route::put(
+            '/registrations/{registration}',
+            [
+            RegistrationController::class,
+            'reject'
+            ]
```
Comment (ghost) [inline routes/web.php]: indent 4 spaces

**Comment by tungbd-0214** [comment]
ready

File: routes/web.php [hunk 13]
```diff
@@ -41,9 +41,17 @@ function () {
             '/registrations',
             [
             RegistrationController::class,
-            'index'
+                'index'
```
Comment (ghost) [inline routes/web.php]:
```suggestion
                'index',
```

File: routes/web.php [hunk 14]
```diff
@@ -41,9 +41,17 @@ function () {
             '/registrations',
             [
             RegistrationController::class,
```
Comment (ghost) [inline routes/web.php]:
```suggestion
                RegistrationController::class,
```

File: routes/web.php [hunk 15]
```diff
@@ -41,9 +41,17 @@ function () {
...
-            'index'
+                'index'
             ]
         )->name('registrations.index');
+
+        Route::put(
+            '/registrations/{registration}',
+            [
+            RegistrationController::class,
+                'reject'
```
Comment (ghost) [inline routes/web.php]:
```suggestion
                RegistrationController::class,
                'reject',
```

**Comment by tungbd-0214** [comment]
ready

**Pull request merged**